docs/developer/buddypress-database-diagram.md

```diff
@@ -0,0 +1,388 @@
+```
+
+```
+       +------------------------+    +---------------------+
+       | wp_bp_xprofile_data     |    | wp_bp_xprofile_fields |
+       +------------------------+    +---------------------+
+       | id                     |    | id                  |
+       | field_id               |    | group_id            |
+       | user_id                |    | parent_id           |
+       | value                  |    | type                |
+       | last_updated           |    | name                |
+       +------------------------+    | description         |
+                                     | is_required         |
+                                     | is_default_option   |
+                                     | field_order         |
+                                     | option_order        |
+                                     | order_by            |
+                                     | can_delete          |
+                                     +---------------------+
+```
+
+```
+       +------------------------+    +------------------------+
+       | wp_bp_xprofile_groups   |    | wp_bp_xprofile_meta     |
+       +------------------------+    +------------------------+
+       | id                     |    | id                     |
+       | name                   |    | object_id              |
+       | description            |    | object_type            |
+       | group_order            |    | meta_key               |
+       | can_delete             |    | meta_value             |
+       +------------------------+    +------------------------+
+```
+
+```
+       +-------------------+    +----------------------+
+       | wp_bp_user_blogs   |    | wp_bp_user_blogs_meta|
+       +-------------------+    +----------------------+
+       | id                |    | id                   |
+       | user_id           |    | blog_id              |
+       | blog_id           |    | meta_key             |
+       +-------------------+    | meta_value           |
+                                 +----------------------+
+```
+
+```
+       +---------------------+    +--------------------+
+       | wp_bp_optouts        |    | wp_bp_signups      |
+       +---------------------+    +--------------------+
+       | id                  |    | signup_id          |
+       | email_address_hash   |    | domain             |
+       | user_id             |    | path               |
+       | email_type          |    | title              |
+       | date_modified       |    | user_login         |
+       +---------------------+    | user_email         |
+                                  | registered         |
+                                  | activated          |
+                                  | active             |
+                                  | activation_key     |
+                                  | meta               |
+                                  +--------------------+
+```
```

Comment:
These are very interesting schema representations. But I believe we should keep it simple. The image would need to be edited for instance.